sys/riscv/riscv/trap.c

```diff
@@ -138,14 +136,18 @@ cpu_fetch_syscall_args(struct thread *td)
 		if (SV_PROC_FLAG(td->td_proc, SV_CHERI))
 			stack_args = (char * __capability)td->td_frame->tf_sp;
 #endif
+	} else {
+		*dst_ap = *ap++;
+		dst_ap++;
```

Comment:
Why not:
```suggestion
		*dst_ap++ = *ap++;
```
? Seems weird to increment one but not the other like that.